File: examples/example-kubernetes-client-openapi-informer-python/src/main/python/micronaut/client/secret/SecretResourceEventHandler.py
# tag::class[]
import logging

from micronaut.kubernetes.client.openapi.informer.handler import Informer, ResourceEventHandler
from micronaut.kubernetes.client.openapi.model import V1Secret
# end::class[]
from micronaut.context.annotation import Requires
from micronaut.context.env import Environment
# tag::class[]

LOG = logging.getLogger(__name__)

# end::class[]

@Requires(env=Environment.KUBERNETES)
# tag::class[]
@Informer(apiType=V1Secret)  # <1>
class SecretResourceEventHandler(ResourceEventHandler[V1Secret]):  # <2>

    def onAdd(self, obj: V1Secret) -> None:
        LOG.info("%s secret added!", obj.getMetadata().getName())

    def onUpdate(self, old_obj: V1Secret, new_obj: V1Secret) -> None:
        LOG.info("%s secret updated!", old_obj.getMetadata().getName())

    def onDelete(self, obj: V1Secret, deleted_final_state_unknown: bool) -> None:
        LOG.info("%s secret deleted!", obj.getMetadata().getName())
# end::class[]

File: examples/example-kubernetes-client-openapi-informer-kotlin/src/main/kotlin/micronaut/client/secret/SecretResourceEventHandler.kt
//tag::class[]
package micronaut.client.secret

import io.micronaut.kubernetes.client.openapi.informer.handler.Informer
import io.micronaut.kubernetes.client.openapi.informer.handler.ResourceEventHandler
import io.micronaut.kubernetes.client.openapi.model.V1Secret
import org.slf4j.LoggerFactory
//end::class[]
import io.micronaut.context.annotation.Requires
import io.micronaut.context.env.Environment

@Requires(env = [Environment.KUBERNETES])
//tag::class[]

@Informer(apiType = V1Secret::class) // <1>
class SecretResourceEventHandler : ResourceEventHandler<V1Secret> { // <2>

    companion object {
        private val LOG = LoggerFactory.getLogger(SecretResourceEventHandler::class.java)
    }

    override fun onAdd(obj: V1Secret) {
        LOG.info("{} secret added!", obj.metadata?.name)
    }

    override fun onUpdate(oldObj: V1Secret, newObj: V1Secret) {
        LOG.info("{} secret updated!", oldObj.metadata?.name)
    }

    override fun onDelete(obj: V1Secret, deletedFinalStateUnknown: Boolean) {
        LOG.info("{} secret deleted!", obj.metadata?.name)
    }
}
//end::class[]
